rclcpp/test/test_parameter_events_filter.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <memory>
+
+#include "rclcpp/exceptions.hpp"
+#include "rclcpp/parameter_client.hpp"
+
+#include "rcl_interfaces/msg/parameter_event.hpp"
+
+class TestParameterEventFilter : public ::testing::Test
+{
+protected:
+  void SetUp()
+  {
+    empty = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    full = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    multiple = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    np = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    cp = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    dp = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+
+    rcl_interfaces::msg::Parameter p;
+    p.name = "new";
+    p.value.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    p.value.integer_value = 1;
+    full->new_parameters.push_back(p);
+    np->new_parameters.push_back(p);
+    multiple->new_parameters.push_back(p);
+
+    p.name = "new2";
+    p.value.integer_value = 2;
+    multiple->new_parameters.push_back(p);
+
+    p.name = "changed";
+    p.value.integer_value = 1;
+    full->changed_parameters.push_back(p);
+    cp->changed_parameters.push_back(p);
+
+    p.name = "deleted";
+    p.value.integer_value = 1;
+    full->deleted_parameters.push_back(p);
+    dp->changed_parameters.push_back(p);
+  }
+
+  // void TearDown()
+  // {
+  // }
+
+  rcl_interfaces::msg::ParameterEvent::SharedPtr empty;
+  rcl_interfaces::msg::ParameterEvent::SharedPtr full;
+  rcl_interfaces::msg::ParameterEvent::SharedPtr multiple;
+  rcl_interfaces::msg::ParameterEvent::SharedPtr np;
+  rcl_interfaces::msg::ParameterEvent::SharedPtr cp;
+  rcl_interfaces::msg::ParameterEvent::SharedPtr dp;
+
+  rclcpp::ParameterEventsFilter::EventType nt = rclcpp::ParameterEventsFilter::EventType::NEW;
+  rclcpp::ParameterEventsFilter::EventType ct = rclcpp::ParameterEventsFilter::EventType::CHANGED;
+  rclcpp::ParameterEventsFilter::EventType dt = rclcpp::ParameterEventsFilter::EventType::DELETED;
+};
+
+/*
+   Testing filters.
+ */
+TEST_F(TestParameterEventFilter, full_by_type) {
+  auto res = rclcpp::ParameterEventsFilter(full,
+      {"new", "changed", "deleted"},
+      {nt, ct, dt});
```

Comment:
nitpick: the indentation here looks like it is one level too many

Reply:
I thought so too, but uncrustify insisted I push it in 4 not 2.

21: --- test/test_parameter_events_filter.cpp
21: +++ test/test_parameter_events_filter.cpp.uncrustify
21: @@ -77,2 +77,2 @@
21: -    {"new", "changed", "deleted"},
21: -    {nt, ct, dt});
21: +      {"new", "changed", "deleted"},
21: +      {nt, ct, dt});
21:

Reply:
The problem is that the code doesn't wrap immediately after the open parenthesis. If you follow that the indentation should be one level only.